File: 1/BubbleSort.java
class BubbleSort {
	
	public static void bubbleSort(int array[]) {
		int n = array.length;

		for(int i=0; i<n-1; i++) {
			boolean flag = true;
			for(int j=0; j<n-i-1; j++) {
				if(array[j+1] < array[j]) {
					flag = false;
					int t = array[j+1];
					array[j+1] = array[j];
					array[j] = t;
				}
			}
			if(flag) {
				break;
			}
		}

	}

	public static void main(String args[]) {
		int array[] = new int[]{8,5,2,9,5,6,3};		
		
		bubbleSort(array);
		
		for(int i : array) {
			System.out.print(String.format("%d \t", i));
		}

		System.out.println();
	}
}

File: 1/js/src/BubbleSort.ts
function swap(i: number, j: number, list: number[]) {
	const temp  = list[i];
	list[i] = list[j];
	list[j] = temp;
}


function bubbleSort(list: number[]) {
	const n = list.length;

	for(let i=0; i<n-1; i++) {
		let flag = true;

		for(let j=0; j<n-i-1; j++) {
			if(list[j] > list[j+1]) {
				swap(j, j+1, list);
				flag = false;
			}
		}

		if(flag) {
			break;
		}
	}

}

(function main() {
	const list = [7,3,5,1,8,4,3];
	bubbleSort(list);
	console.log(list);
})();
